resolve/index.go

```diff
@@ -189,7 +196,8 @@ func (ix *RuleIndex) collectEmbeds(r *ruleRecord) {
 			continue
 		}
 		ix.collectEmbeds(er)
-		if resolver == ix.mrslv(er.rule, er.file.Pkg) {
```

Comment:
Matching the names of the resolvers would be sufficient.

Per the doc on `Resolver.Embeds()`:

	// Embeds returns a list of labels of rules that the given rule embeds. If
	// a rule is embedded by another importable rule of the same language, only
	// the embedding rule will be indexed. The embedding rule will inherit
	// the imports of the embedded rule.

So as long as the languages are the same, it's embeded.

And per `Resolver.Name()`:

	// Name returns the name of the language. This should be a prefix of the
	// kinds of rules generated by the language, e.g., "go" for the Go extension
	// since it generates "go_library" rules.

Hence, comparing the names will do the trick.

Reply:
To be more specific, all we need is to check if the languages are the same, not the resolvers. The current implementation uses a stronger assertion, which implies that the languages are the same.

Reply:
Yes, it does the job. Thx